tests/unit/cluster/slot-migration.tcl

```diff
@@ -435,3 +435,36 @@ start_cluster 2 0 {tags {tls:skip external:skip cluster regression} overrides {c
         R 0 MIGRATE 127.0.0.1 [lindex [R 1 CONFIG GET port] 1] $stream_name 0 5000
     }
 }
+
+start_cluster 3 6 {tags {external:skip cluster} overrides {cluster-node-timeout 1000} } {
+    test "Killing all replicas in primary 0" {
+        assert_equal 2 [s 0 connected_slaves]
+        catch {R 3 shutdown nosave}
+        catch {R 6 shutdown nosave}
+        wait_for_condition 50 100 {
+            [s 0 connected_slaves] == 0
+        } else {
+            fail "The replicas in primary 0 are still connecting"
+        }
+    }
+
+    test "Killing one replica in primary 1" {
+        assert_equal 2 [s -1 connected_slaves]
+        catch {R 4 shutdown nosave}
+        wait_for_condition 50 100 {
+            [s -1 connected_slaves] == 1
+        } else {
+            fail "The replica in primary 1 is still connecting"
+        }
+    }
+
```

Comment:
I think same comment as I gave in another PR. tests should be independent so I suggest move that logic to the  'test "Slot migration is ok when the replicas are down"'

Reply:
right, i forget this one, i updated it. 

Actually we have some other old tests, like replication part is doing this old way, a separate test is used to start the replication and other tests do the rest things.